src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -139,5 +139,100 @@ public static async Task TestBOMWithNoJsonValue()
                     async () => await JsonSerializer.DeserializeAsync<byte>(stream));
             }
         }
+
+        public static IEnumerable<object[]> BOMWithJsonElementTestData
+        {
+            get
+            {
+                yield return CreateStreamArgs(100);
+                yield return CreateStreamArgs(200);
+                yield return CreateStreamArgs(400);
+                yield return CreateStreamArgs(800);
+                yield return CreateStreamArgs(1600);
+
+                static object[] CreateStreamArgs(int count)
+                {
+                    byte[] objBytes = Encoding.UTF8.GetBytes(
+                        @"{""Test"":{},""Test2"":[],""Test3"":{""Value"":{}},""PersonType"":0,""Id"":2}");
+
+                    byte[] comma = Encoding.UTF8.GetBytes(@",");
+
+                    var stream = new MemoryStream();
+
+                    // Write BOM.
+                    stream.Write(Encoding.UTF8.GetPreamble());
```

Comment:
All these calls to write are failing to compile for desktop since it is missing the span overloads.